### PR TITLE
Sample details refactor/ update sample file seq objects

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Updated UI for displaying sample files in the Files tab of the SampleDetailViewer. This update allows us to use this component wherever sequencing objects / genome assemblies need to be listed.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Review by joshsadam — CHANGES_REQUESTED** [review]
Quick review before I leave for the day.  This one should not take long to get in.

File: src/main/webapp/resources/js/components/sequence-files/GenomeAssemblyListItem.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export function GenomeAssemblyListItem({ genomeAssembly, actions = [] }) {
+  const { label, createdDate } = genomeAssembly.fileInfo;
+  return (
+    <div
+      style={{
+        display: "flex",
+        justifyContent: "space-between",
+        alignItems: "center",
+        width: `100%`,
+        border: "solid 1px #EEE",
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/GenomeAssemblyListItem.jsx:23]: Still need to update the colors.. just wanted you to take a look and see what you think. I will get this and the comment above about the colors updated

File: src/main/webapp/resources/js/components/samples/components/GenomeAssemblyList.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const { sample, modifiable: isModifiable } = useSelector(
+    (state) => state.sampleReducer
+  );
+  const { files } = useSelector((state) => state.sampleFilesReducer);
+
+  /*
+   Download genome assembly files
+   */
+  const downloadAssemblyFile = ({ sampleId, genomeAssemblyId }) => {
+    notification.success({
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/GenomeAssemblyList.jsx:25]: The message just shows that the download has started not that it has completed successfully

File: src/main/webapp/resources/js/components/sequence-files/SequenceObjectListItem.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  };
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        justifyContent: "space-between",
+        alignItems: "center",
+        width: `100%`,
+        border: "solid 1px #EEE",
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/SequenceObjectListItem.jsx]: Updated in ef966187d0dfd7cb4ca35752a639b85f30ec3ec1

File: src/main/webapp/resources/js/components/sequence-files/SequenceObjectListItem.jsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        padding: 5,
+      }}
+    >
+      {displayConcatenationCheckbox !== null && (
+        <div style={{ marginTop: sequenceObject.qcEntries?.length ? -50 : 0 }}>
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/SequenceObjectListItem.jsx]: Added in ef966187d0dfd7cb4ca35752a639b85f30ec3ec1

File: src/main/webapp/resources/js/components/sequence-files/SequenceObjectListItem.jsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+          ) : (
+            <FileOutlined />
+          )
+        }
+      />
+      <List style={{ width: `100%` }} itemLayout="horizontal" component="div">
+        <List.Item
+          actions={actions}
+          className="t-file-details"
+          key={
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/SequenceObjectListItem.jsx]: Redone in ef966187d0dfd7cb4ca35752a639b85f30ec3ec1

File: src/main/webapp/resources/js/components/samples/components/GenomeAssemblyList.jsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    downloadGenomeAssemblyFile({ sampleId, genomeAssemblyId });
+  };
+
+  /*
+   Get the actions required for a Genome Assembly
+   */
+  const getActionsForGenomeAssembly = (genomeAssembly) => {
+    let actions = [];
+
+    actions.push(
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/GenomeAssemblyList.jsx]: Added to array declaration in ef966187d0dfd7cb4ca35752a639b85f30ec3ec1

File: src/main/webapp/resources/js/components/sequence-files/GenomeAssemblyListItem.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/GenomeAssemblyListItem.jsx]: Updated in ef966187d0dfd7cb4ca35752a639b85f30ec3ec1

[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/sequence-files/GenomeAssemblyListItem.jsx]: See comment above

**Comment by joshsadam** [comment]
This looks so much better, thank you. 2 suggestions:
1. Add a little padding on both sides of the avatars, it looks crowded.
2. I know why you put file size at the end, I like the way it looks, it does break the fact that the action buttons should always be at the end. What if you were to make the action buttons a fixed width, and then put the qc and files size before them?

**Comment by joshsadam** [comment]
Sorry, one other thing I just noticed.  After uploading a file, I know you are querying the server for the qc status, but once it is good it seems to keep hitting  the server:
image

**Comment by deepsidhu85** [comment]
> Thanks. I don't know if `grey4` would be too dark, but there is a default `BORDER_LIGHT` constant in borders.js.

Nope it still looks great. Updated in d9ac9542450c9f3fb277d4bfee8e658a96d2cbde

**Comment by deepsidhu85** [comment]
> Sorry, one other thing I just noticed. After uploading a file, I know you are querying the server for the qc status, but once it is good it seems to keep hitting the server: image

I have not been able to recreate this even after trying with different number of files per upload. If there's anything specific that you did let me know and I can try out the same steps

**Comment by deepsidhu85** [comment]
> This looks so much better, thank you. 2 suggestions:
> 
> 1. Add a little padding on both sides of the avatars, it looks crowded.
> 2. I know why you put file size at the end, I like the way it looks, it does break the fact that the action buttons should always be at the end. What if you were to make the action buttons a fixed width, and then put the qc and files size before them?

Added a marginLeft to the avatars (d9ac9542450c9f3fb277d4bfee8e658a96d2cbde) which doesn't make it look as crowded. I tried putting a padding on but it shifts the icon inside the avatar

**Comment by deepsidhu85** [comment]
> This looks so much better, thank you. 2 suggestions:
> 
> 1. Add a little padding on both sides of the avatars, it looks crowded.
> 2. I know why you put file size at the end, I like the way it looks, it does break the fact that the action buttons should always be at the end. What if you were to make the action buttons a fixed width, and then put the qc and files size before them?

Moved action buttons to the end in 66af80a9f4bd217959db51fd3b6dcc4232640fc2.

**Review by joshsadam — CHANGES_REQUESTED** [review]
Just that one last change

**Pull request merged**